homeworks/feliistvanattila/8_streamlit/feliia_weather_app.py

```diff
@@ -0,0 +1,112 @@
+import streamlit as st
+import pandas as pd
+from pandas import DataFrame
+import numpy as np
+import datetime
+import requests
+from requests import Response
+
+
+class FetchWeatherResult:
+    def __init__(self):
+        self.status_code: int
+        self.dataFrame: DataFrame
+        self.lon: float
+        self.lat: float
+        self.temperature: str
+        self.humidity: str
+        self.wind_speed: str
+        self.no_data_found: bool
+        self.error_text: str
+        self.response: Response
+
+
+API_KEY = st.secrets["openweather"]["api_key"]
+
+
+# @st.cache_data(ttl=86400)
+def fetch_weather_data(city) -> FetchWeatherResult:
+
+    result = FetchWeatherResult()
+
+    weather_url = f'http://api.openweathermap.org/data/2.5/weather?q={
+        city}&appid={API_KEY}&units=metric'
+
+    response = requests.get(url=weather_url)
+
+    if response.status_code == 200:
+        result.response = response
```

Comment:
a result.response-t minden ágban megadod, szóval igazából ezt igazából ki is lehetne vinni az if-else elé, és akkor nem kell háromszor leírni.